src/main/java/com/uid2/admin/vertx/service/SharingService.java

```diff
@@ -150,6 +156,66 @@ private void handleSetKeyset(RoutingContext rc) {
         }
     }
 
+    // Method to check if one set contains any values from another set
+    private static <T> boolean containsAny(Set<T> set1, Set<T> set2) {
+        for (T element : set2) {
+            if (set1.contains(element)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private void handleListAllKeysetsRelated(RoutingContext rc) {
+        try {
+            // Get value for site id
+            final Optional<Integer> siteIdOpt = RequestUtil.getSiteId(rc, "site_id");
+            if (!siteIdOpt.isPresent()) return;
+            final int siteId = siteIdOpt.get();
+
+            if (siteId != Const.Data.AdvertisingTokenSiteId && !SiteUtil.isValidSiteId(siteId)) {
+                ResponseUtil.error(rc, 400, "must specify a valid site id");
+                return;
+            }
+
+            // Get value for client type
+            Set<ClientType> clientTypes = new HashSet<>();
+            if (!rc.queryParam("client_types").isEmpty()) {
+                clientTypes = getTypes(rc.queryParam("client_types").get(0));
+            }
+
+//            // Check if the key has a ID_READER role
+            boolean isIdReaderRole = true;
```

Comment:
I set this to constant `true`. Question here is: if we need to rotate related keysets for a participant, do we care whether the leaked key is ID_READER or not? 

My take is we will just rotate regardless of type of the leaked key. WDYT?